presto-docs/src/main/sphinx/connector/pinot.rst

```diff
@@ -0,0 +1,105 @@
+===============
+Pinot Connector
+===============
+
+The Pinot connector allows Presto to query data stored in
+`Apache Pinot™ (Incubating) <https://pinot.apache.org/>`_.
+
+Compatibility
+-------------
+
+The Pinot connector is compatible with all Pinot versions starting from 0.1.0.
+
+Configuration
+-------------
+
+To configure the Pinot connector, create a catalog properties file
+``etc/catalog/pinot.properties`` with atleast the following contents,
+replacing ``host1:9000,host2:9000`` with a comma-separated list of Pinot Controller nodes
+
+.. code-block:: none
+
+    connector.name=pinot
+    pinot.controller-urls=host1:9000,host2:9000
+
+Configuration Properties
+------------------------
+
+The following configuration properties are available:
+
+================================================== ========== ===================================================================================
+Property Name                                      Required   Description
+================================================== ========== ===================================================================================
+``pinot.controller-urls``                          Yes        A comma separated list of controller hosts. If Pinot is deployed via
+                                                              `Kubernetes <https://kubernetes.io/>`_ this needs to point to the controller
+                                                              service endpoint. The Pinot broker and server must be accessible via dns as
+                                                              Pinot will return hostnames and not ip addresses.
+``pinot.segments-per-split``                       No         The number of segments processed in a split. Setting this higher will reduce
+                                                              the number of requests made to Pinot which makes sense for a smaller Pinot cluster.
+``pinot.request-timeout``                          No         The timeout for Pinot requests. Increasing this can reduce timeouts if dns
+                                                              resolution is slow.
+================================================== ========== ===================================================================================
+
+Querying Pinot Tables
+-------------------------
+
+The schema for each Pinot catalog is ``default``. To find all tables you can run::
+
+    SHOW TABLES FROM pinot.default;
+
+To see a list of columns in the ``flight_status`` table the following can be run:
+
+    DESCRIBE pinot.default.flight_status;
+    SHOW COLUMNS FROM pinot.default.flight_status;
+
+Tables can be queried as they are in other connectors, leveraging filter and limit pushdown:
```

Comment:
Queries written with SQL are fully supported and can include filters and limits::